elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -213,48 +178,66 @@ private Object fetchObject(RequestScope requestScope, Class recordType, Optional
             Optional<FilterExpression> filterExpression;
             EntityDictionary dictionary = requestScope.getDictionary();
 
-            Class<?> idType = dictionary.getIdType(recordType);
-
-            String idField = dictionary.getIdFieldName(recordType);
-
-            String entityTypeName = dictionary.getJsonAliasFor(recordType);
+            Class<?> idType = dictionary.getIdType(entityClass);
+            String idField = dictionary.getIdFieldName(entityClass);
+            String entityTypeName = dictionary.getJsonAliasFor(entityClass);
 
             /* construct a new SQL like filter expression, eg: book.id IN [1,2] */
             filterExpression = Optional.of(new FilterPredicate(
                     new FilterPredicate.PathElement(
-                            recordType,
+                            entityClass,
                             entityTypeName,
                             idType,
                             idField),
                     Operator.IN,
                     new ArrayList<>(idList)));
-
-            return PersistentResource.loadRecords(recordType, requestScope, filterExpression);
+            return PersistentResource.loadRecords(entityClass, requestScope, filterExpression);
         }
     }
 
     /**
-     * FETCH attributes of top level entity
-     * @param parentResource parent object
-     * @param field Field type
-     * @return list of persistent resource objects
+     * fetches a non-root level attribute or relationship
+     * @param dictionary entity dictionary
+     * @param ids list of ids
+     * @param parentResource parent resource
+     * @param fieldName field type
+     * @return attribute or relationship object
      */
-    private Object fetchAttribute(PersistentResource parentResource, Field field) {
-        return parentResource.getAttribute(field.getName());
+    private Object fetchObject(EntityDictionary dictionary, Optional<List<String>> ids,
```

Comment:
It looks like this is only called in one place.  I think we should fold this into the calling function.  It is a function that does two things and takes arguments that don't apply for one of those two things.